web/packages/teleport/src/Roles/RoleEditor/RoleEditor.tsx

```diff
@@ -144,7 +157,12 @@ export const RoleEditor = ({
     // requires model to be valid. However, if it's OK, we reset the validator.
     // We don't want it to be validating at this point, since the user didn't
     // attempt to submit the form.
-    if (!standardModel.roleModel.requiresReset && !validator.validate()) return;
+    if (
+      standardModel.roleModel !== undefined &&
+      !standardModel.roleModel.requiresReset &&
+      !validator.validate()
```

Comment:
Just a theoretical question rather than a request because I don't have too much time to dive deep into this, but would it be possible to organize the code in such a way that we don't have to explicitly check for `standardModel.roleModel` being undefined?

I see it's repeated in a couple of places, I guess the reason behind it is that `!standardModel.roleModel?.requiresReset` would have different, unwanted semantics.

But it's not something that's done often and I feel like it's going to throw the reader of this code for a spin whenever they encounter one of those checks. So if it'd be possible to simplify it, then it'd only be better.

I suppose changing the underlying object shape might be difficult, but then maybe encapsulating this in a function would work?

Reply:
Hmm. Actually, I was considering replacing the `requiresReset` field with a function that would return a boolean looking at the number of conversion errors. So I'll do exactly this.

Reply:
Okay, not exactly. There are three places in which I explicitly compare the model to `undefined`, and in each case, the logic is slightly different. I don' think that extracting it to a function brings us any closer to this being more understandable.